WaterLog_Backend/Procedures.cs

```diff
@@ -50,9 +50,9 @@ public async Task TriggerInsert(ReadingsEntry value)
 
             if (IsLeakage(reading1.Value, reading2.Value))
             {
+                await CreateSegmentsEventAsync(segmentid, "leak", reading1.Value, reading2.Value);
                 //Updateleakagestatus
-                IEnumerable<SegmentLeaksEntry> allLeaks = _db.SegmentLeaks;
-                if (allLeaks.Any(leak => leak.SegmentsId == segmentid))
+                if (await _db.SegmentLeaks.AnyAsync(leak => leak.SegmentsId == segmentid && leak.ResolvedStatus == EnumResolveStatus.UNRESOLVED))
```

Comment:
I think extract the await into a variable above the if and use that variable in the if statement